src/chart/rendering/drawitem.cpp

```diff
@@ -71,45 +70,73 @@ void drawItem::drawLines(const Styles::Guide &style,
 
 void drawItem::draw()
 {
-	if (!shouldDraw()) return;
+	if (!shouldDrawMarkerBody()) return;
 
 	if (drawOptions.onlyEssentials() && (double)plot.anySelected
 	    && (double)marker.selected == 0)
 		return;
 
-	auto lineFactor = options.shapeType.factor<Math::FuzzyBool>(
-	    Gen::ShapeType::line);
-
-	auto circleFactor = options.shapeType.factor<Math::FuzzyBool>(
-	    Gen::ShapeType::circle);
-
-	if (lineFactor != false && circleFactor != false) {
+	if (options.shapeType.contains(Gen::ShapeType::line) 
+		&& options.shapeType.contains(Gen::ShapeType::circle))
+	{
 		CircleItem circle(marker,
 		    coordSys,
 		    options, 
 		    plot.getStyle());
 
-		LineItem line(marker,
-		    coordSys,
-		    options,
-		    plot.getStyle(),
-		    plot.getMarkers(),
-		    0);
-
 		draw(circle, 1, false);
-		draw(line, 1, true);
-	}
-	else {
-		auto blended0 = DrawItem::createInterpolated(marker,
-		    options,
-		    plot.getStyle(),
-		    coordSys,
-		    plot.getMarkers(),
-		    0);
 
-		double lineFactorD = static_cast<double>(lineFactor);
-		draw(blended0, (1 - lineFactorD) * (1 - lineFactorD), false);
-		draw(blended0, sqrt(lineFactorD), true);
+		marker.prevMainMarkerIdx.visit([&, this](int index, auto value){
+			ConnectingItem line(marker,
+					coordSys,
+					options,
+					plot.getStyle(),
+					plot.getMarkers(),
+					index,
+					Gen::ShapeType::line);
+
+			draw(line, value.weight, true);
+		});
+	}
+	else 
+	{
+		auto drawMarker = [&, this](int index, ::Anim::Weighted<uint64_t> value) {
+			auto blended0 = DrawItem::createInterpolated(marker,
+				options,
+				plot.getStyle(),
+				coordSys,
+				plot.getMarkers(),
+				index);
+	
+			auto lineFactor = 
+				options.shapeType.factor<double>(Gen::ShapeType::line);
+
+			draw(blended0, value.weight * (1 - lineFactor) * (1 - lineFactor), false);
+			draw(blended0, value.weight * sqrt(lineFactor), true);
+		};
+
+		auto containsConnected = 
+			options.shapeType.contains(Gen::ShapeType::line) 
+			|| options.shapeType.contains(Gen::ShapeType::area);
+
+		auto containsSingle = 
+			options.shapeType.contains(Gen::ShapeType::rectangle) 
+			|| options.shapeType.contains(Gen::ShapeType::circle);
+
+		if (containsConnected)
+		{
+			if (containsSingle) 
+			{
+				auto lineIndex = 
+					options.shapeType.get(0).value == Gen::ShapeType::line 
```

Comment:
Is this not the `Vizzu::Gen::isConnecting`?

Reply:
Yes, it is.